src/LocalFeedConfigs.php

```diff
@@ -0,0 +1,119 @@
+<?php //phpcs:disable WordPress.WP.AlternativeFunctions --- Uses FS read/write in order to reliable append to an existing file.
+/**
+ * Class responsible for managing local feed configurations.
+ *
+ * @package     Pinterest_For_WooCommerce/Classes/
+ * @version     x.x.x
+ */
+namespace Automattic\WooCommerce\Pinterest;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Handling feed files generation.
+ *
+ * Singleton pattern used.
+ * Prevent application from having multiple instances of configuration.
+ * At the same time allow distributed use of configurations.
+ */
+class LocalFeedConfigs {
+
+	/**
+	 * Array of local feed configurations.
+	 *
+	 * @var array $feeds_configurations
+	 */
+	private $feeds_configurations = null;
+
+	/**
+	 * The Singleton's instance.
+	 *
+	 * @since x.x.x
+	 * @var null|LocalFeedConfigs Instance object.
+	 */
+	private static $instance = null;
+
+	/**
+	 * Singleton initialization and instance fetching method.
+	 *
+	 * @since x.x.x
+	 * @return LocalFeedConfigs Singleton instance.
+	 */
+	public static function get_instance(): LocalFeedConfigs {
+		$cls = static::class;
+		if ( null === self::$instance ) {
+			self::$instance = new static();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Class responsible for local feed configurations and handling.
+	 *
+	 * @since x.x.x
+	 */
+	protected function __construct() {
+		$locations = array( Pinterest_For_Woocommerce()::get_base_country() ?? 'US' ); // Replace with multiple countries array for multiple feed config.
```

Comment:
I think that we will not use filter here. Here is why.

For now we only use what is set in `get_base_country` or use default `US`.
When multifeed capability will be added the country list will be handled from the UI.
The UI part is an assumption that it will be up to the merchant, and should be easy to configure by them.

Reply:
I am marking as resolved but please reopen in case you think that the filter is needed.